code/game/machinery/GBPvendor.dm

```diff
@@ -0,0 +1,125 @@
+		new /datum/GBP_equipment("Cigar",				/obj/item/clothing/mask/cigarette/cigar/havana,						50,	1),
+		new /datum/GBP_equipment("Soap",				/obj/item/weapon/soap/nanotrasen,									100,	1),
+		new /datum/GBP_equipment("Insulated Gloves",				/obj/item/clothing/gloves/color/yellow,					100,	1),
+		new /datum/GBP_equipment("Fulton Beacon",		/obj/item/fulton_core,												50,	1),
+		new /datum/GBP_equipment("Fulton Pack",			/obj/item/weapon/extraction_pack,									200,	1),
+		new /datum/GBP_equipment("Space Cash",			/obj/item/stack/spacecash/c1000,									250,	1),
+		new /datum/GBP_equipment("50 metal sheets",			/obj/item/stack/sheet/metal/fifty,								500,	1),
+		new /datum/GBP_equipment("50 glass sheets",			/obj/item/stack/sheet/glass/fifty,								500,	1),
+		new /datum/GBP_equipment("50 cardboard sheets",			/obj/item/stack/sheet/cardboard/fifty,						500,	1),
+		new /datum/GBP_equipment("Hardsuit x3",			/obj/item/clothing/suit/space/hardsuit,								750,	3),
+		new /datum/GBP_equipment("Jetpack Upgrade x5",		/obj/item/weapon/tank/jetpack/suit,								1000,	5),
+		new /datum/GBP_equipment("Advanced Magboot x5",			/obj/item/clothing/shoes/magboots/advance,					2000,	5),
+		new /datum/GBP_equipment("ERT Hardsuit x5",		/obj/item/clothing/suit/space/hardsuit/ert,							6000,	5),
+		new /datum/GBP_equipment("Portal Gun x5",			/obj/item/weapon/gun/energy/wormhole_projector,					6000,	5),
+		new /datum/GBP_equipment("Reactive Decoy Armor x5",		/obj/item/clothing/suit/armor/reactive/stealth,				7500,	5),
+		new /datum/GBP_equipment("Cloaking Belt x5",		/obj/item/device/shadowcloak,									8000,	5),
+		new /datum/GBP_equipment("Chrono Suit x5",		/obj/item/clothing/suit/space/chronos,								10000,	5),
+		new /datum/GBP_equipment("WHAT HAVE YOU DONE... x5",		/obj/vehicle/space/speedbike/speedwagon,				30000,	5),
+		)
+
+/datum/GBP_equipment
+	var/equipment_name = "generic"
+	var/equipment_path = null
+	var/cost = 0
+	var/amount = 0
+
+/datum/GBP_equipment/New(name, path, cost, amount)
+	src.equipment_name = name
+	src.equipment_path = path
+	src.cost = cost
+	src.amount = amount
+
+/obj/machinery/GBP_vendor/power_change()
+	..()
+	update_icon()
+
+/obj/machinery/GBP_vendor/update_icon()
+	if(powered())
+		icon_state = initial(icon_state)
+	else
+		icon_state = "[initial(icon_state)]-off"
+
+/obj/machinery/GBP_vendor/attack_hand(mob/user)
+	if(..())
+		return
+	interact(user)
+
+/obj/machinery/GBP_vendor/interact(mob/user)
+	GBPearned = GBP + GBPspent
+	var/dat
+	dat +="<div class='statusDisplay'>"
+	dat += "You have <td>[GBP]</td> engineering voucher points<br>"
+	switch(round(GBPearned*100/world.time))
+		if(0 to 5)
+			dat += "Rating: Terrible<br>"
+		if(6 to 10)
+			dat += "Rating: Bad<br>"
+		if(11 to 15)
+			dat += "Rating: Subpar<br>"
+		if(16 to 20)
+			dat += "Rating: Decent<br>"
+		if(21 to 25)
+			dat += "Rating: Robust<br>"
+		if(26 to 50)
+			dat += "Rating: Good boy<br>"
+		if(51 to 999)
+			dat += "Rating: Holy Shit<br>"

```

Comment:
This is another placeholder for a concept I'm going to flesh out more, maybe even try to incorporate stats into it.